pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -217,19 +217,60 @@ func (c *Cluster) startMons(targetCount int) error {
 		return fmt.Errorf("failed to assign pods to mons. %+v", err)
 	}
 
+	// The centralized mon config database can only be used if there is at least one mon
+	// operational. If we are starting mons, and one is already up, then there is a cluster already
+	// created, and we can immediately set values in the config database. The goal is to set configs
+	// only once and do it as early as possible in the mon orchestration.
+	setConfigsNeedsRetry := false
+	if existingCount > 0 {
+		err := config.SetDefaultAndUserConfigs(c.context, c.Namespace, c.ClusterInfo, c.spec.ConfigOverrides)
+		if err != nil {
+			// If we fail here, it could be because the mons are not healthy, and this might be
+			// fixed by updating the mon deployments. Instead of returning error here, log a
+			// warning, and retry setting this later.
+			setConfigsNeedsRetry = true
+			logger.Warningf("failed to set Rook and/or user-defined Ceph config options before starting mons; will retry after starting mons. %+v", err)
+		}
+	}
+
 	if existingCount < len(mons) {
 		// Start the new mons one at a time
 		for i := existingCount; i < targetCount; i++ {
 			if err := c.ensureMonsRunning(mons, i, targetCount, true); err != nil {
 				return err
 			}
+
+			// If this is the first mon being created, we have to wait until it is created to set
+			// values in the config database. Do this only when the existing count is zero so that
+			// this is only done once when the cluster is created.
+			if existingCount == 0 {
+				err := config.SetDefaultAndUserConfigs(c.context, c.Namespace, c.ClusterInfo, c.spec.ConfigOverrides)
+				if err != nil {
+					return fmt.Errorf("failed to set Rook and/or user-defined Ceph config options after creating the first mon. %+v", err)
```

Comment:
This could mean the flag is not settable during runtime, so perhaps we should indicate which flag failed to be set and recommend using the ConfigMap
`rook-config-override`?